aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -36,10 +36,14 @@ pub static APTOS_TRANSACTION_VALIDATION: Lazy<TransactionValidation> =
         module_addr: CORE_CODE_ADDRESS,
         module_name: Identifier::new("transaction_validation").unwrap(),
         fee_payer_prologue_name: Identifier::new("fee_payer_script_prologue").unwrap(),
+        fee_payer_prologue_v2_name: Identifier::new("fee_payer_script_prologue_v2").unwrap(),
```

Comment:
We can upgrade private functions iirc, so this code can be avoided

Reply:
if we update the function arg list inline, then the rust side needs an additional feature flag to know when to include the `required_deposit` arg, which i thought is worse than defining a func v2 + using v2 only when `required_deposit.is_some` (so no additional feature flag is needed)?


cc @zekun000

Reply:
cc @msmouse @grao1991

Reply:
I thought we need a flag anyway, we can't use required_deposit to decide if calling v1 or v2 because the framework may not be updated yet